Update crate documentation (#104)
bitcoindevkit/bdk-cli: README.md

    <a href="https://github.com/bitcoindevkit/bdk-cli/blob/master/LICENSE"><img alt="MIT or Apache-2.0 Licensed" src="https://img.shields.io/badge/license-MIT%2FApache--2.0-blue.svg"/></a>
    <a href="https://github.com/bitcoindevkit/bdk-cli/actions?query=workflow%3ACI"><img alt="CI Status" src="https://github.com/bitcoindevkit/bdk-cli/workflows/CI/badge.svg"></a>
    <a href="https://codecov.io/gh/bitcoindevkit/bdk-cli"><img src="https://codecov.io/gh/bitcoindevkit/bdk-cli/branch/master/graph/badge.svg"/></a>
    <a href="https://docs.rs/bdk-cli"><img alt="API Docs" src="https://img.shields.io/badge/docs.rs-bdk_cli-green"/></a>
    <a href="https://blog.rust-lang.org/2020/08/27/Rust-1.56.0.html"><img alt="Rustc Version 1.56+" src="https://img.shields.io/badge/rustc-1.56%2B-lightgrey.svg"/></a>
    <a href="https://discord.gg/d7NkDKm"><img alt="Chat on Discord" src="https://img.shields.io/discord/753336465005608961?logo=discord"></a>
  </p>

  <h4>
    <a href="https://bitcoindevkit.org">Project Homepage</a>
    <span> | </span>
    <a href="https://docs.rs/bdk-cli">Documentation</a>
  </h4>
</div>

## About

> This project provides a command-line Bitcoin wallet application using the latest [BDK APIs](https://docs.rs/bdk/0.19.0/bdk/wallet/struct.Wallet.html). This is used mostly as an high level integration testing framework, by the BDK team, using the [tests](/tests/integration.rs) modules.

Reviewer: ```suggestion
This project provides a command-line Bitcoin wallet application using the latest [BDK APIs](https://docs.rs/bdk/0.19.0/bdk/wallet/struct.Wallet.html). This interface is used mostly as a framework for high level integration testing, by the BDK team, using the [tests](/tests/integration.rs) modules.
```

> But if you are planning to use BDK in your own wallet project, bdk-cli is a nice playground to get started with. It allows easy testnet and regtest wallet operations, to try out whats possible with descriptors, miniscripts, and BDK APIs. For more information on BDK refer the [website](https://bitcoindevkit.org/), and the [rust docs](https://docs.rs/bdk/latest/bdk/index.html)

Reviewer: ```suggestion
If you consider using BDK in your own wallet project, bdk-cli is a nice playground to get started with. It allows easy testnet and regtest wallet operations, to try out whats possible with descriptors, miniscripts, and BDK APIs. For more information on BDK refer the [website](https://bitcoindevkit.org/), and the [rust docs](https://docs.rs/bdk/latest/bdk/index.html)
```

> bdk-cli can be compiled in various different ways to suit the experimental needs.

Reviewer: bdk-cli can be compiled in various different ways to suit the experimental needs.

> bdk-cli can be compiled in various different ways to suit the experimental needs.

Reviewer: ```suggestion
bdk-cli can be compiled with different features to suit the experimental needs.
```

  - Database Options
     - `key-value-db` : Sets the wallet database a `sled` db.
     - `sqlite-db` : Sets the wallet database as `sqlite3` db.
  - Blockchain Options
     - `rpc` : Connects the wallet to bitcoin core via RPC.
     - `electrum` : Connects the wallet to an electrum server.
     - `compact_filters` : Deploy a BIP157 node to get blockchain data from bitcoin p2p network.
     - `esplora-ureq/reqwest` : Connects the wallet to a esplora server sync/asynchronously.
  - Extra Utility Tools
     - `repl` : use bdk-cli as a [REPL](https://codewith.mu/en/tutorials/1.0/repl) shell (useful for quick hand testing wallet operations).
     - `compiler` : opens up bdk-cli policy compiler commands.
     - `verify` : uses `bitcoinconsensus` to verify transactions at every `sync` call of the wallet.
     - `reserves` : opens up bdk-cli **Proof of Reserves** operation commands using the [bdk-reserves plugin](https://github.com/weareseba/bdk-reserves). (requires `electrum`)
   - Automated Node Backend

>      - `regtest-bitcoin` : Auto deploys a regtest bitcoin node, connects the wallet, and exposes core rpc commands via `bdk-cli node` subsommands.

Reviewer: ```suggestion
     - `regtest-bitcoin` : Auto deploys a regtest bitcoin node, connects the wallet, and exposes core rpc commands via `bdk-cli node` subcommands.
```

>      - `regtest-electrum` : Auto deploys a `electrsd` connected to a `bitcoind` and an wallet connectded to `electrsd`. `bdk-cli node` subcommand still calls the `bitcoind` RPC.

Reviewer: ```suggestion
     - `regtest-electrum` : Auto deploys a `electrsd` connected to a `bitcoind` and a wallet connectded to `electrsd`. `bdk-cli node` subcommand still calls the `bitcoind` RPC.
```
